iocore/net/UnixNetAccept.cc

```diff
@@ -251,13 +251,23 @@ NetAccept::init_accept_per_thread()
   }
 }
 
+//
+// Stop NetAccept and its slave objects
+//
 void
 NetAccept::stop_accept()
 {
-  if (!action_->cancelled) {
-    action_->cancel();
+  forl_LL(NetAccept, na, slaves)
+  {
+    na->ep.stop();
   }
-  server.close();
+  ep.stop();
+
+  NetAccept *na;
+  while ((na = slaves.pop())) {
+    na->cancel();
+  }
+  cancel();
```

Comment:
Since you add slaves level here, you may need to walk through all the children's slave to avoid the mistake by other people.

Reply:
No one calls `NetAccept::cancel()` directly. The `server.fd` is added into each of ET_NET Threads and should do the `ep.stop()` to remove `server.fd` from each of ET_NET Threads.

Any cloned `NetAccept` is a slave and will be attached to the master `NetAccept`. Only the master `NetAccept` is pushed into the `naVec` queue.

Slaves share `server.fd` with the master, or open a new `server.fd` if it is necessary (with SO_REUSEPORT).

Reply:
It seems to be that `stop_accept` calls `ep.stop` and `cancel` on the NA, and on slaves, therefore you could have one loop here that calls `stop_accept` on the slaves, which would then do the `ep.stop` and `cancel`.

I wonder if the queue should be in the `NAVec`, e.g. make that a vector of lists.